packages/ruleset/src/functions/string-boundary.js

```diff
@@ -20,68 +25,74 @@ const bypassFormats = {
   maxLength: ['date']
 };
 
+const compositionFields = ['allOf', 'anyOf', 'oneOf'];
+
 function stringBoundaryErrors(schema, path) {
+  // We don't want to flag a violation for a (potentially) partial schema
+  // defined in an allOf/anyOf/oneOf list element.
+  if (path.length >= 2 && compositionFields.includes(path[path.length - 2])) {
+    return [];
+  }
```

Comment:
I *think* you can handle this by just using `validateNestedSchemas` instead of `validateSubschemas`

Reply:
I'll change this...

Reply:
I verified that using validateNestedSchemas works and I can remove that if-block.